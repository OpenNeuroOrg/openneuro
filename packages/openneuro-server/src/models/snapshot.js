import mongoose from 'mongoose'

const snapshotSchema = new mongoose.Schema({
  datasetId: { type: String, required: true },
  tag: { type: String, required: true },
  created: { type: Date, default: Date.now },
  hexsha: { type: String },
})

snapshotSchema.index({ datasetId: 1, tag: 1 }, { unique: true })

const Snapshot = mongoose.model('Snapshot', snapshotSchema)

export default Snapshot
