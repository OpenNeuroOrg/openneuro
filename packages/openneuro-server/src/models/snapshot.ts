import mongoose from "mongoose"
import type { Document } from "mongoose"
const { Schema, model } = mongoose

export interface SnapshotDocument extends Document {
  id: string
  datasetId: string
  tag: string
  created: Date
  hexsha: string
}

const snapshotSchema = new Schema({
  datasetId: { type: String, required: true },
  tag: { type: String, required: true },
  created: { type: Date, default: Date.now },
  hexsha: { type: String },
})

snapshotSchema.index({ datasetId: 1, tag: 1 }, { unique: true })
snapshotSchema.index({ datasetId: 1 })

const Snapshot = model<SnapshotDocument>("Snapshot", snapshotSchema)

export default Snapshot
