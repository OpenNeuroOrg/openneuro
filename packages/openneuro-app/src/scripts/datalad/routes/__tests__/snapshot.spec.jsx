import React from 'react'
import { shallow } from 'enzyme'
import { NoErrors } from '../snapshot.jsx'

describe('NoErrors component in SnapshotRoute', () => {
  it('handles extant errors case', () => {
    const issues = [{ severity: 'error' }]
    expect(
      shallow(
        <NoErrors issues={issues}>
          <div>child</div>
        </NoErrors>,
      ),
    ).toMatchSnapshot()
  })
  it('handles no authors case', () => {
    const issues = [{ severity: 'warning', code: 113, key: 'NO_AUTHORS' }]
    expect(
      shallow(
        <NoErrors issues={issues}>
          <div>child</div>
        </NoErrors>,
      ),
    ).toMatchSnapshot()
  })
  it('handles case of errors and no authors', () => {
    const issues = [
      { severity: 'error' },
      { severity: 'warning', code: 113, key: 'NO_AUTHORS' },
    ]
    expect(
      shallow(
        <NoErrors issues={issues}>
          <div>child</div>
        </NoErrors>,
      ),
    ).toMatchSnapshot()
  })
  it('returns children when issues are okay', () => {
    const issues = []
    expect(
      shallow(
        <NoErrors issues={issues}>
          <div>child</div>
        </NoErrors>,
      ),
    ).toMatchSnapshot()
  })
})
