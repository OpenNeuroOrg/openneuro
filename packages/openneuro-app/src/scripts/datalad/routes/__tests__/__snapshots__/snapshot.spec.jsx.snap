// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoErrors component in SnapshotRoute handles case of errors and no authors 1`] = `
<span
  className="text-danger"
>
  BIDS validation must be complete and all errors corrected and "Authors" must include at least one entry in dataset_description.json to create a snapshot
</span>
`;

exports[`NoErrors component in SnapshotRoute handles extant errors case 1`] = `
<span
  className="text-danger"
>
  BIDS validation must be complete and all errors corrected to create a snapshot
</span>
`;

exports[`NoErrors component in SnapshotRoute handles no authors case 1`] = `
<span
  className="text-danger"
>
  "Authors" must include at least one entry in dataset_description.json to create a snapshot
</span>
`;

exports[`NoErrors component in SnapshotRoute returns children when issues are okay 1`] = `
<div>
  child
</div>
`;
